app/src/Liquidator.ts

```diff
@@ -0,0 +1,236 @@
+    );
+    this.txManager = new TxManager(this.web3);
+    this.borrowers = [];
+    this.uniqueId = Math.floor(100000 + Math.random() * 900000).toString();
+    this.limiter = new Bottleneck({
+      minTime: MS_BETWEEN_REQUESTS,
+    });
+    this.errorCount = 0;
+  }
+
+  public start() {
+    winston.log("info", `🔋 Powering up liquidation bot #${this.uniqueId}`);
+    this.txManager.init();
+
+    this.collectBorrowers(ALOE_INITIAL_DEPLOY);
+
+    this.pollingInterval = setInterval(() => {
+      console.log("Scanning borrowers...");
+      this.scan(this.borrowers);
+    }, POLLING_INTERVAL_MS);
+  }
+
+  public async stop(): Promise<void> {
+    winston.log("info", `🪫 Powering down liquidation bot #${this.uniqueId}`);
+    if (this.pollingInterval) {
+      clearInterval(this.pollingInterval);
+    }
+    return new Promise((resolve, reject) => {
+      this.web3.eth.clearSubscriptions((error: Error, result: boolean) => {
+        if (error) {
+          console.error(error);
+          return reject(error);
+        }
+        return resolve();
+      });
+    });
+  }
+
+  public async isHealthy(): Promise<HealthCheckResponse> {
+    if (this.errorCount > ERROR_THRESHOLD) {
+      return {
+        code: STATUS_NOT_HEALTHY,
+        message: "Liquidator error threshold exceeded",
+      };
+    }
+    try {
+      const result: boolean = await this.web3.eth.net.isListening();
+      console.log("Is listening?", result);
+      if (!result) {
+        return {
+          code: STATUS_NOT_HEALTHY,
+          message: "unable to listen to peers",
+        };
+      }
+    } catch (e) {
+      const msg: string = (e as Error).message;
+      return {
+        code: STATUS_NOT_HEALTHY,
+        message: msg,
+      };
+    }
+    if (!this.txManager.isHealthy()) {
+      return {
+        code: STATUS_NOT_HEALTHY,
+        message: "transaction manager is not healthy",
+      };
+    }
+    return {
+      code: STATUS_HEALTHY,
+      message: "healthy",
+    };
+  }
+
+  private collectBorrowersCallback(error: Error, result: Log) {
+    if (!error) {
+      const borrowerAddress: string = Liquidator.formatAddress(result.data);
+      if (!this.borrowers.includes(borrowerAddress)) {
+        winston.log(
+          "debug",
+          `Detected new borrower! Adding \`${borrowerAddress}\` to global list (${this.borrowers.length} total).`
+        );
+        this.borrowers.push(borrowerAddress);
+      } else {
+        winston.log(
+          "debug",
+          `Received duplicate creation event for borrower ${borrowerAddress}`
+        );
+      }
+    } else {
+      this.errorCount += 1;
+      winston.log("error", `Error when collecting borrowers: ${error}`);
+    }
+  }
+
+  private collectBorrowers(block: number) {
+    this.web3.eth.subscribe(
+      "logs",
+      {
+        address: FACTORY_ADDRESS,
+        topics: [CREATE_ACCOUNT_TOPIC_ID],
+        fromBlock: block,
+      },
+      this.collectBorrowersCallback.bind(this)
```

Comment:
Good call on the `.bind(this)` -- I had a lot of trouble with forgetting that in my old bots.